eth/executionclient/execution_client.go

```diff
@@ -249,11 +288,33 @@ func (ec *ExecutionClient) Healthy(ctx context.Context) error {
 		return ErrClosed
 	}
 
+	lastHealthyTime := time.Unix(ec.lastSyncedTime.Load(), 0)
+
+	if ec.healthInvalidationInterval != 0 && time.Since(lastHealthyTime) <= ec.healthInvalidationInterval {
+		// Synced recently, reuse the result (only if ec.healthInvalidationInterval is set).
+		return nil
+	}
+
+	err := ec.is_healthy(ctx)
+	if err != nil {
+		close(ec.healthy)
+	} else {
+		// Reset the healthy channel if it was closed.
+		select {
+		case <-ec.healthy:
+		default:
+			ec.healthy = make(chan struct{})
+		}
+	}
+	return errors.Join(ErrUnhealthy, err)
+}
```

Comment:
i think if we add lock & defer unlock at the beginning of the func, it should solve the parallel request issue

the first parallel caller with lock and update time, the other caller will wait for lock and see that time updated and return immediately

Reply:
done